### PR TITLE
clang-tidy: Enable readability-container-size-empty check

### PR DESCRIPTION
**Pull request by masterleinad**
We started making sure to use `empty()` on container types and it turns out that there is a `clang-tify` check to ensure that. This pull request enables that.

### CODE REVIEW

File: cmake/scripts/expand_instantiations.cc [hunk 1]
```diff
@@ -100,12 +100,12 @@ std::string
...
 {
   std::string x;
-  while (in.size() != 0)
+  while (!in.empty())
     {
       // stop copying to the result if the current character is a
       // delimiter, but only if the previous character was not a backslash
       if ((delim_list.find(in[0]) != std::string::npos) &&
-          !((x.size() > 0) && (x[x.size() - 1] == '\\')))
+          !((!x.empty()) && (x[x.size() - 1] == '\\')))
```
Comment (bangerth) [inline cmake/scripts/expand_instantiations.cc:108]: That's unreadable -- a double negation. There needs to be a better way.

**Review by bangerth — COMMENTED** [review]
I don't like this. I find `v.size()>0` to be much easier to read than `!v.empty()`. `v.empty()` is a negative statement: "Nothing is in the vector". `!v.empty()` is a double negative. It's hard to read.

I have no objection changing `v.size() == 0` to `v.empty()`.

**Comment by bangerth** [comment]
What do we want to about this patch? I'm ok with all the places where you replace `v.size() == 0` by `v.empty()` (and would accept a patch that takes these parts of your diffs), but I find reading `v.size() > 0` much easier than `!v.empty()`.

**Comment by masterleinad** [comment]
> What do we want to about this patch? I'm ok with all the places where you replace `v.size() == 0` by `v.empty()` (and would accept a patch that takes these parts of your diffs), but I find reading `v.size() > 0` much easier than `!v.empty()`.

I believe all those places have been covered in other pull requests. Let's simply close this one then.

**Pull request closed without merging**